lib/mayaUsd/fileio/jobs/writeJob.cpp

```diff
@@ -296,85 +371,142 @@ bool UsdMaya_WriteJob::_BeginWriting(const std::string& fileName, bool append)
         if (status != MS::kSuccess) {
             continue;
         }
+        if (curDagPathStr.empty())
+            continue;
 
         argDagPaths.insert(curDagPathStr);
 
         status = curDagPath.pop();
+
         if (status != MS::kSuccess) {
             continue;
         }
         curDagPathIsValid = curDagPath.isValid(&status);
 
         while (status == MS::kSuccess && curDagPathIsValid) {
             curDagPathStr = curDagPath.partialPathName(&status).asChar();
+
             if (status != MS::kSuccess) {
-                break;
+                goto ctn;
```

Comment:
That `goto` is going to make a lot of people uncomfortable. The use is correct, though recommended only when exiting nested loops. Please review http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-goto and potentially the next recommendation regarding breaking out of loops.
In doubt, just leave the code as is.